config/prod.exs

```diff
@@ -6,4 +6,5 @@ config :logger, :splunk_log,
   host: "https://http-inputs-mbta.splunkcloud.com/services/collector/event",
   token: {:system, "SOCKET_PROXY_SPLUNK_TOKEN"},
   format: "$dateT$time $metadata[$level] node=$node $message\n",
-  level: :info
+  level: :info,
+  max_buffer: 1
```

Comment:
Are there any performance implications to not buffering? I'm remembering when realtime_signs went down because we were using the old version of the Splunk library that didn't buffer and the requests started taking slightly too long.

Reply:
From a performance perspective, yeah, it's certainly not great to not buffer. That said, the old socket proxy was using the older version of `logger_splunk_backend` that didn't buffer, and never had the same problems `realtime_signs` did, that I can recall. I think it just doesn't log enough information for it to matter.

The trade off is in how long we want to wait for Splunk to see the updates. I bumped it down to `1` from its default of `32` because we were only seeing batch logs appear every half hour. In the normal case, it only logs `ehmon` heartbeat once/min, unless there are disconnects/reconnects.

We have a Splunk "heartbeat" alert that looks for the `ehmon` logs, so I didn't want false positives. I could potentially bump the buffer to, like, `5`, and then add a 5 minute delay to the heartbeat. But in terms of performance gains from a small buffer vs. a 5 minute delay in alerting, "the juice didn't seem worth the squeeze" to me.

Reply:
Makes sense. It sounds like the issues with the alert is complex enough that it isn't worth addressing right now. I do sort of worry, though, since realtime_signs was also using the non-buffering version for a long time without issue. 😅 Maybe something to address later in a different ticket.

Reply:
Yeah, I certainly don't love the approach. It would be nifty to update `logger_splunk_backend` to skip the buffer if the last message was a long time ago. As you say, we can watch out for issues and address later if need be.